src/supervisor/watchers/handlers/index.ts

```diff
@@ -69,7 +69,7 @@ const workloadWatchMetadata: Readonly<IWorkloadWatchMetadata> = {
       paginatedReplicationControllerList(namespace),
   },
   [WorkloadKind.CronJob]: {
```

Comment:
I wonder if we need to introduce a new workload kind - CronJobV1, and the existing one becomes CronJobV1Beta1. This way we can run both readers side by side, otherwise we will only be able to detect one but not the other

Reply:
but this is based on kind which is `CronJob` for both!

Reply:
the v1 reader will work for v1 and v1beta1 CronJobs (verified by fixture in integration test)